modules/currency-l0/src/main/scala/org/tessellation/currency/l0/Main.scala

```diff
@@ -1,86 +1,72 @@
-package org.tessellation.currency
+package org.tessellation.currency.l0
+
+import java.util.UUID
 
 import cats.effect.{IO, Resource}
-import cats.syntax.applicative._
-import cats.syntax.applicativeError._
 import cats.syntax.semigroupk._
 
-import org.tessellation.currency.cli.method
-import org.tessellation.currency.cli.method.{Run, RunGenesis, RunValidator}
-import org.tessellation.currency.http.P2PClient
-import org.tessellation.currency.modules._
-import org.tessellation.currency.schema.currency.{CurrencySnapshot, TokenSymbol}
+import org.tessellation.currency.l0.cli.method
+import org.tessellation.currency.l0.cli.method.{Run, RunGenesis, RunValidator}
+import org.tessellation.currency.l0.http.P2PClient
+import org.tessellation.currency.l0.modules._
+import org.tessellation.currency.schema.currency.CurrencySnapshot
+import org.tessellation.currency.{BuildInfo, CurrencyKryoRegistrationIdRange, currencyKryoRegistrar}
 import org.tessellation.ext.cats.effect.ResourceIO
 import org.tessellation.ext.kryo._
-import org.tessellation.schema.address.Address
 import org.tessellation.schema.cluster.ClusterId
 import org.tessellation.schema.node.NodeState
 import org.tessellation.sdk.app.{SDK, TessellationIOApp}
-import org.tessellation.sdk.cli.CliMethod
-import org.tessellation.sdk.domain.collateral.OwnCollateralNotSatisfied
 import org.tessellation.sdk.infrastructure.genesis.{Loader => GenesisLoader}
 import org.tessellation.sdk.infrastructure.gossip.{GossipDaemon, RumorHandlers}
 import org.tessellation.sdk.resources.MkHttpServer
 import org.tessellation.sdk.resources.MkHttpServer.ServerName
 import org.tessellation.sdk.{SdkOrSharedOrKernelRegistrationIdRange, sdkKryoRegistrar}
-import org.tessellation.security.signature.Signed
 
 import com.monovore.decline.Opts
-import eu.timepit.refined.auto._
 import eu.timepit.refined.boolean.Or
 
-abstract class CurrencyApp[A <: CliMethod](
-  header: String,
-  clusterId: ClusterId,
-  helpFlag: Boolean,
-  version: String,
-  symbol: TokenSymbol,
-  identifier: Address
-) extends TessellationIOApp[A](clusterId.toString, header, clusterId, helpFlag, version) {
+object Main
+    extends TessellationIOApp[Run](
+      "Currency-l0",
+      "Currency L0 node",
+      ClusterId(UUID.randomUUID()),
```

Comment:
I guess this will prevent the nodes from connecting with each other, because for every node the clusterId will be different right? 🤔

Reply:
That of course is the case if we are checking the clusterId during join, and I think we are.

Reply:
@TheMMaciek i think it works on compile time but let's confirm @kpudlik. If it runs on runtime then @TheMMaciek is right

Reply:
You are right. I was debugging something so I forced different values for ClusterId and forgot to revert it to fixed one.

Reply:
Of course it still works if you have just one L0 node

Reply:
fixed